Issue #12 by gkapoor121212: Add controller for contribution …

gkapoor121212 opened the pull request:
…statistics.

hussainweb commented:
I am considering if we should add more stuff to `contrib_tracker`. I was thinking more about removing the module entirely over time. That said, this change feels like it should be in this module.

Do you have any thoughts on how we would remove deprecated code from this module over time?

gkapoor121212 commented:
@hussainweb We can add a new module 'ct_reports' which can have all the logic for statistical data and contribution count. The Drupal console command (if it is still used) can be moved to  'ct_manager' module.

hussainweb commented:
I suppose we will have more reports in the future and so it makes sense. The Drupal console command is not used anymore. It was a one-time thing to clean up a bug in the past.

gkapoor121212 commented:
@hussainweb This is available for review now. I have removed code from the contrib_tracker module and added a new ct_reports module. For now, I have added a new service that has methods that return statistical data and a controller to display those.

hussainweb commented on web/modules/custom/ct_reports/src/Controller/ContributionCountController.php:
```diff
@@ -0,0 +1,53 @@
+      '#codeContributions' => $this->contribStats->codeContributions(),
+      '#totalContributors' => $this->contribStats->totalContributors(),
+    ];
+    return $build;
```
@gkapoor121212, let's add reasonable caching.

gkapoor121212 replied:
@hussainweb I searched a little a bit and found this [changelog](https://www.drupal.org/node/3107058). So for now I have set a tag 'node_list:code_contribution'.

hussainweb commented on web/modules/custom/ct_reports/ct_reports.info.yml:
```diff
@@ -0,0 +1,4 @@
+name: Contribution Tracker Reports
+type: module
+description: Provides functionality to generate reports based on contribution data.
+core_version_requirement: ^8.9 || ^9
```
```suggestion
core_version_requirement: ^9
```

Let's skip supporting Drupal 8.9.
